File: 1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
        stack<char> st;
        int maxDepth=0;
        for(char ch:s)
        {
            if(ch=='(') st.push(ch);
            else if(ch==')') st.pop();
            maxDepth=max(maxDepth,(int)st.size());
        }
        return maxDepth;
    }
};

File: 1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
        int n = s.length();
        int count = 0, maxDepth = 0;

        for(int i=0;i<n;i++) {
            char ch = s.charAt(i);
            if(ch == '(') count++;
            else if(ch == ')') count--;
            maxDepth = Math.max(maxDepth,count);
        }
        return maxDepth;
    }
}